gluten-core/src/main/scala/io/glutenproject/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -166,23 +166,36 @@ case class FilterExecTransformer(condition: Expression, child: SparkPlan)
         substraitContext, condition, child.output, operatorId, null, validation = true)
     } catch {
       case e: Throwable =>
-        logValidateFailure(
-          s"Validation failed for ${this.getClass.toString} due to ${e.getMessage}", e)
+        this.appendValidateLog(s"Validation failed for ${this.getClass.toString}" +
```

Comment:
It looks the constructing of arg for `appendValidateLog` can be simplified since many are in similar pattern,
e.g., 
```
def appendValidateLog(failureMsg) {
 log = "${this.getClass.toString}: validation failed, reason: failureMsg"
 validateLog = validateLog :+ log
}
```

Reply:
Not all of them are in this case and we use keyword `due to` for both java and native side.

Reply:
@PHILO-HE This is a API in `trait GlutenPlan` and it's not quite necessary to add another similar API in it. But I have consolidate the ones in `TransformHintRule` and seems cleaner. Please take a further look.